app/code/Transoft/Blog/ViewModel/Bloglist.php

```diff
@@ -0,0 +1,81 @@
+<?php
+declare(strict_types=1);
+namespace Transoft\Blog\ViewModel;
+
+use Magento\Catalog\Block\Product\View;
+use Magento\Framework\View\Element\Block\ArgumentInterface;
+use Magento\Framework\App\ObjectManager;
+use Transoft\Blog\Helper\Data;
+use Transoft\Blog\Model\ResourceModel\Post\Collection;
+use Transoft\Blog\Model\ResourceModel\Post\CollectionFactory as PostCollectionFactory;
+
+/**
+ * Class Bloglist
+ *
+ * Blog list view model class
+ */
+class Bloglist implements ArgumentInterface
+{
+    /**
+     * @var PostCollectionFactory
+     */
+    protected $modelFactory;
+
+    /**
+     * Transoft\Blog\Helper\Data $helper
+     */
+    protected $helper;
+
+    /**
+     * Constructor.
+     *
+     * @param PostCollectionFactory $modelFactory
+     * @param Data $helper
+     */
+    public function __construct(
+        PostCollectionFactory $modelFactory,
+        Data $helper
+    ) {
+        $this->modelFactory = $modelFactory;
+        $this->helper = $helper;
+    }
+
+    /**
+     * Returns model collection
+     *
+     * @return Collection
+     */
+    public function retrieveModel()
+    {
+        $collection = $this->modelFactory->create();
+        $collection->setOrder('creation_time', 'ASC')->setPageSize(5);
+
+        return $collection;
+    }
+
+    /**
+     * Check if types matching
+     *
+     * @return bool
+     */
+    public function isTypesMatching()
+    {
+        $types = $this->helper->getConfig('catalog/blog/blog_applied_to');
+        $product = ObjectManager::getInstance()->get(View::class);
```

Comment:
Why do we need to get Magento\Catalog\Block\Product\View block just to get *current_product*?

Reply:
I use this because all registry related methods are deprecated. This one is the only alternative I've found.

Reply:
Please **don't do** like this. Please use for ex.: \Magento\Catalog\Model\Locator\LocatorInterface::getProduct

Reply:
fixed